### PR TITLE
Refactor query types and update zdk

### PR DESCRIPTION
**Pull request by iainnash**

### CODE REVIEW

File: src/backends/opensea/OpenseaDataSource.ts [hunk 1]
```diff
@@ -23,14 +23,13 @@ export class OpenseaDataSource implements OpenseaInterface {
...
-    return await this.nftsLoader.load(`${tokenContract}:${tokenId}`);
+  loadNFT = async ({ contract, id }: NFTIdentifier): Promise<OpenseaAsset | Error> => {
+    return await this.nftsLoader.load(`${contract}:${id}`);
   };
-  loadNFTs = async (tokenContractAndId: string[]): Promise<(OpenseaAsset | Error)[]> => {
-    return await this.nftsLoader.loadMany(tokenContractAndId);
+  loadNFTs = async (
+    nfts: readonly NFTIdentifier[]
+  ): Promise<(OpenseaAsset | Error)[]> => {
+    return await this.nftsLoader.loadMany(nfts.map((nft) => `${nft.contract}-${nft.id}`));
```
[the remark this answers is not in the document]
Comment (iainnash) [reply src/backends/opensea/OpenseaDataSource.ts:32]: We should make a constant for this in NFT_INTERFACE perhaps NFT_ID_SEPERATOR ?

File: src/backends/zdk-alpha/ZDKAlphaDataSource.ts [hunk 2]
```diff
@@ -104,7 +103,7 @@ function getMarkets(markets: TokenMarketResponseItem['markets']) {
   const marketResponse: MarketModule[] = [];
   markets.forEach((market) => {
     const getStandardMarketData = (
-      market: TokenMarketResponseItem['markets'][0],
+      market: MarketInfoFragment & MarketDetailsFragment,
```
Comment (iainnash) [inline src/backends/zdk-alpha/ZDKAlphaDataSource.ts:106]: Is there a reason why this changed from TokenResponseItem['markets'][0]? This works, but another option is to create a new type.

File: src/backends/zdk-alpha/ZDKAlphaDataSource.ts [hunk 3]
```diff
@@ -308,36 +307,34 @@ export class ZDKAlphaDataSource implements ZDKAlphaDataInterface {
...
 
-  loadNFT = async (
-    tokenContract: string,
-    tokenId: string
-  ): Promise<TokenMarketResponseItem | Error> => {
-    const response = await this.zdk.tokenMarkets({
+  loadNFT = async ({
+    contract,
+    id,
+  }: NFTIdentifier): Promise<TokenResponseItem | Error> => {
```
Comment (iainnash) [inline src/backends/zdk-alpha/ZDKAlphaDataSource.ts:313]: nice :)

File: src/backends/zora-graph/ZoraGraphDataSource.ts [hunk 4]
```diff
@@ -93,15 +99,17 @@ export class ZoraGraphDataSource implements ZoraGraphDataInterface {
...
     }
-    return await this.nftGraphDataLoader.load(tokenId);
+    return await this.nftGraphDataLoader.load(id);
   };
 
-  async loadNFTs(tokenContractAndId: readonly string[]) {
-    return await this.nftGraphDataLoader.loadMany(tokenContractAndId);
+  async loadNFTs(nfts: readonly NFTIdentifier[]) {
+    return await this.nftGraphDataLoader.loadMany(
+      nfts.map(({ contract, id }) => getAddress(`${contract}:${id}`))
```
Comment (iainnash) [inline src/backends/zora-graph/ZoraGraphDataSource.ts:111]: this should be:
`${getAddress(contract)}${SEPERATOR}${id}` getAddress will break if you pass it `:${id}` it just needs the contract address.

File: src/backends/zora-indexer-v1/ZoraIndexerV1DataSource.ts [hunk 5]
```diff
@@ -455,11 +456,14 @@ export class ZoraIndexerV1DataSource implements ZoraIndexerV1Interface {
...
-    return await this.nftGraphDataLoader.load(`${getAddress(tokenContract)}-${tokenId}`);
+  loadNFT = async ({ contract, id }: NFTIdentifier) => {
+    return await this.nftGraphDataLoader.load(`${getAddress(contract)}-${id}`);
   };
-  loadNFTs = async (tokenContractAndId: readonly string[]) => {
-    return await this.nftGraphDataLoader.loadMany(tokenContractAndId);
+
+  loadNFTs = async (nfts: readonly NFTIdentifier[]) => {
+    return await this.nftGraphDataLoader.loadMany(
+      nfts.map((nft) => `${getAddress(nft.contract)}-${nft.id}`)
```
Comment (iainnash) [inline src/backends/zora-indexer-v1/ZoraIndexerV1DataSource.ts:465]: note: this is the correct `getAddress()`

**Review by iainnash — COMMENTED** [review]
Left a few comments ~

File: src/backends/zora-indexer-v1/ZoraIndexerV1DataSource.ts (hunk 5 above)
Comment (olegakbarov) [reply src/backends/zora-indexer-v1/ZoraIndexerV1DataSource.ts]: +

File: src/backends/zora-graph/ZoraGraphDataSource.ts (hunk 4 above)
Comment (olegakbarov) [reply src/backends/zora-graph/ZoraGraphDataSource.ts]: +

File: src/backends/zdk-alpha/ZDKAlphaDataSource.ts (hunk 2 above)
Comment (olegakbarov) [reply src/backends/zdk-alpha/ZDKAlphaDataSource.ts]: Fixed

File: src/backends/opensea/OpenseaDataSource.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (olegakbarov) [reply src/backends/opensea/OpenseaDataSource.ts]: Done

**Review by iainnash — COMMENTED** [review]
Looks great!

**Pull request merged**